### PR TITLE
Close All Toasts Feature

### PR DESCRIPTION
**Pull request by choeqq**
### Issues 😢 

- #193

---

### What has been done 😄 

- [x] Created close all toasts feature - unless there is only one toast, the button will always be displayed on last toast visible
- [x] animations are little off, each toast should go down one by one (done ✅ )

---

### Screenshots 📸 
Screenshot 2023-10-21 at 20 39 58 copy

### CODE REVIEW

**Review by choeqq — COMMENTED** [review]
Hi! Just updated Close All feature, please check out if you like it @lucasspra 😉

**Comment by emilkowalski** [comment]
> Hi! Just updated `Close All Feature`, please check out whether you like it or not @lucasspra 😉

This is great, can we center the text in the button vertically and move the text `closeAllButtonText` to `<Toaster />` instead?

File: pnpm-lock.yaml
Comment (emilkowalski) [inline pnpm-lock.yaml:1]: Can we exclude this file from this PR?

File: src/styles.css [hunk 1]
```diff
@@ -44,6 +47,8 @@ html[dir='rtl'],
   --gray11: hsl(0, 0%, 43.5%);
   --gray12: hsl(0, 0%, 9%);
   --border-radius: 8px;
+  --button-box-shadow: box-shadow: 0px 4px 12px rgba(0, 0, 0, 0.1),
```
Comment (emilkowalski) [inline src/styles.css:50]: Typo

**Review by choeqq — COMMENTED** [review]
All requested changes has been commited, please let me know whether you have any other ideas to make this solution greater 😄 

Since all the changes has been applied I allowed myself to resolve your comments, I don't know how it works in this project yet - so please hit me up if that's not the way you all are doing it right here 😄 

@emilkowalski 👀

**Review by choeqq — COMMENTED** [review]
@emilkowalski could you please take a look? 😄 🚀

**Comment by emilkowalski** [comment]
@choeqq Sorry for the delay, it looks great! Only missing thing is exit animation on the first toast, it just disappears. Could you take a look at it and then we can merge?

https://github.com/emilkowalski/sonner/assets/36730035/9ffcc675-9477-4740-be4b-0ac2f6bc6593

**Comment by choeqq** [comment]
No problem, everyone of us is busy 😄 
The flow is as follows: after clicking `close all` we're invoking the 'folding' of all toasts just like when hover out of toasts, and then they just dismiss, I think it looks fine, but you're the owner of the project so, please feel free to add any requriements, and I'll adjust my solution to your needs 😄

**Comment by emilkowalski** [comment]
Also, we should add documentation as well.

**Comment by choeqq** [comment]
Hi @emilkowalski just added API reference for close all toasts feature, I am having little trouble with animating the fadeout of all toasts, could you please help me little bit with that part? 😄

**Comment by andersonbruno** [comment]
Was this removed? I noticed in the last push to main, that it isn't there anymore. Any chance to have this functionality?

**Comment by jihea-park** [comment]
@emilkowalski 
Was this removed??

**Comment by tp0ns** [comment]
Do we have context on why it was removed ? Is there any plan to add this feature back ?

**Comment by hskang9** [comment]
Why is this removed? is there a way I can get this feature back?

**Comment by FathZulfaAli** [comment]
Can we have this feature back ? would be helpful to user to dismiss all toaster
Thanks

**Pull request closed without merging**